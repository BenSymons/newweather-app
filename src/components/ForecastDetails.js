import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-details">
      <div className="forecast-details_date">Date:{formattedDate}</div>
      <div className="forecast-details_humidity">Humidity:{humidity}</div>
      <div className="forecast-details_temperaturemax">
        {" "}
        Max Temp:
        {temperature.max}&deg;C
      </div>
      <div className="forecast-details_temperaturemin">
        Min Temp:
        {temperature.min}&deg;C
      </div>
      <div className="forecast-details_windspeed">Wind Speed: {wind.speed}</div>
      <div className="forecast-details_winddirection">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}
export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      direction: PropTypes.string,
      speed: PropTypes.number,
    }),
  }).isRequired,
};
