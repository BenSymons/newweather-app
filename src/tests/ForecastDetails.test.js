import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "800",
    humidity: 14,
    temperature: {
      max: 22,
      min: 12,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);
    expect(getByText("Date:Thu Jan 01 1970")).toHaveClass(
      "forecast-details_date"
    );
    expect(getByText("Min Temp:12°C")).toHaveClass(
      "forecast-details_temperaturemin"
    );
    expect(getByText("Max Temp:22°C")).toHaveClass(
      "forecast-details_temperaturemax"
    );
    expect(getByText("Humidity:14")).toHaveClass("forecast-details_humidity");
    expect(getByText("Wind Speed: 13")).toHaveClass(
      "forecast-details_windspeed"
    );
    expect(getByText("Wind Direction: s")).toHaveClass(
      "forecast-details_winddirection"
    );
  });
});
