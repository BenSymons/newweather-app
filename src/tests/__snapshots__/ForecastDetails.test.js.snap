// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details_date"
    >
      Date:Thu Jan 01 1970
    </div>
    <div
      class="forecast-details_humidity"
    >
      Humidity:14
    </div>
    <div
      class="forecast-details_temperaturemax"
    >
       Max Temp:22°C
    </div>
    <div
      class="forecast-details_temperaturemin"
    >
      Min Temp:12°C
    </div>
    <div
      class="forecast-details_windspeed"
    >
      Wind Speed: 13
    </div>
    <div
      class="forecast-details_winddirection"
    >
      Wind Direction: s
    </div>
  </div>
</DocumentFragment>
`;
